### PR TITLE
testing: add filecheck test for matrix addition and multiplication as target for riscv codegen

### PR DESCRIPTION
**Pull request by superlopuh**
Add some inputs in scf with an integration test for the result. The aim of the current riscv sprint is to be able to lower the module to riscv, print the assembly, execute it on riscemu, and get the correct filecheck print at the end.

Putting up as draft to pull out the changes that I needed to make for the test to work into separate PRs.

@tobiasgrosser @kingiler @adutilleul @nazavode @AntonLydike

### CODE REVIEW

File: xdsl/parser/core.py [hunk 1]
```diff
@@ -730,7 +731,8 @@ def parse_operation(self) -> Operation:
         if (n_bound_results != 0) and (len(op.results) != n_bound_results):
             self.raise_error(
                 f"Operation has {len(op.results)} results, "
-                f"but was given {n_bound_results} to bind."
+                f"but was given {n_bound_results} to bind.",
```
Comment (superlopuh) [inline xdsl/parser/core.py:734]: @math-fehr by the time this error gets printed the current token location points to the next operation, I got quite confused for a bit :)

File: tests/dialects/test_memref.py [hunk 2]
```diff
@@ -427,3 +428,19 @@ def test_memref_copy():
         match="Expected source and destination to have the same element type.",
     ):
         copy.verify()
+
+
+def test_memref_get_global():
```
Comment (superlopuh) [inline tests/dialects/test_memref.py:433]: TODO: move this test to interpreter test

File: tests/interpreters/test_memref_interpreter.py [hunk 3]
```diff
@@ -48,3 +56,26 @@ def test_functions():
...
+        memref.Global.get(
+            StringAttr("my_global"),
+            memref_type,
+            DenseIntOrFPElementsAttr.from_list(tensor_type, [1, 2, 3, 4]),
+            sym_visibility=StringAttr("public"),
+        )
+        with ImplicitBuilder(func.FuncOp("main", ((), ())).body):
+            fetch = memref.GetGlobal.get("my_global", memref_type)
+
+    # assert SymbolTable.lookup_symbol(fetch, "my_global") is define
```
Comment (kingiler) [inline tests/interpreters/test_memref_interpreter.py:75]: Why a commented assertion?

**Comment by superlopuh** [comment]
We're now going for just the GEMM, will close this, and reopen separate PRs for GEMM kernels and the memref global interpreter function

**Pull request closed without merging**